src/pages/MoviePost/MoviePost.jsx

```diff
@@ -1,7 +1,319 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import Posts from './Posts';
+import PostModal from './PostModal';
+import backimg from '../../image/sdf.jpeg';
+import styled from 'styled-components';
+
+const LIMIT = 4;
+// const TOTALCOUNT = 12;
 
 function MoviePost() {
-  return <div>MoviePost</div>;
+  const [rankMovies, setRankMoives] = useState([]);
+  const [movieList, setMovieList] = useState([]);
+  const [modalPost, setModalPost] = useState([]);
+  const [showPostModal, setShowPostModal] = useState(false);
+  const [pagination, setPagination] = useState({
+    offset: 0,
+    limit: LIMIT,
+    totalPage: 0,
+    currentPage: 1,
+  });
+
+  const allRankMovies = rankMovies.map((movie, index) => (
+    <Poster img={movie.poster_image}>
+      <PosterNum>{index + 1}</PosterNum>
+      <AllPostText>
+        <PosterText>{movie.movie_name}</PosterText>
+        <PostText>POST</PostText>
+        <PostCount>{movie.post_count}</PostCount>
+      </AllPostText>
+    </Poster>
+  ));
+
+  const onMoreClick = () => {
+    setPagination({
+      ...pagination,
+      currentPage: pagination.currentPage + 1,
+      offset: pagination.offset + pagination.limit,
+    });
+
+    console.log(pagination);
+  };
+
+  // useEffect(() => {
+  //   if (pagination.offset !== 0)
+  //     fetch(
+  //       `http://localhost:3000/data/page.json?limit=${pagination.limit}&offset=${pagination.offset}`
+  //     )
+  //       .then(response => response.json())
+  //       .then(result => {
+  //         setMovieList(...movieList, result['results']);
+  //         setPagination({
+  //           ...pagination,
+  //           totalPage: Math.ceil(result['totalCount'] / pagination.limit),
+  //         });
+  //       });
+  // }, [pagination.offset]);
+  // const moreClick = () => {
+  //   // setOffset();
+  //   if currentPage < totalPage
+  //   setOffset({ offset: offset + limit });
+  // };
+
+  // useEffect(() => {
+  //   set그거(그거.concat(
+  //     [<그거 / >,<그거 / >....]
+
+  //   ))
+
+  // }, [offset]);
+
+  useEffect(() => {
+    fetch('http://localhost:3000/data/harry.json')
+      .then(response => response.json())
+      .then(data => {
+        setRankMoives(data.MOVIE_LIST);
+      });
+  }, []);
+
+  useEffect(() => {
+    fetch('http://localhost:3000/data/page.json')
+      .then(response => response.json())
+      .then(data => {
+        setMovieList(data);
+      });
+  }, []);
+
+  const clickPost = (e, data) => {
+    setModalPost(data);
+    handleModal();
+
+    console.log(modalPost);
+  };
+
+  const onClickWindow = e => {
+    const clicked = e.target.closest('.modal');
+
+    if (clicked) {
+      return;
+    }
+    handleModal();
+  };
+
+  const handleModal = () => {
+    setShowPostModal(!showPostModal);
+  };
```

Comment:
```suggestion
  const toggleModal = () => {
    setShowPostModal(prev => !prev);
  };
```
- 최대한 함수 외부의 값보다는 내부의 값을 참조하는 방식으로 코딩해주세요
- 구현상 이름이 toggle에 더 맞는 것 같고, 만약 닫는 동작만 하게 된다면 closeModal로 하고 false 값을 아예 주는게 좋겠습니다

Reply:
넵 ! 감사합니다!!